server: customize tracking tables with a custom name (#5719)
hasura/graphql-engine: docs/graphql/core/api-reference/schema-metadata-api/table-view.rst

      }
   }

> An ``identifier`` can be used to track an table with the identifier. This can

Reviewer: ```suggestion
An ``identifier`` can be used to track a table with the identifier. This can
```

> be useful when a table name is not GraphQL compliant, like ``"Users Address"``

Reviewer: ```suggestion
be useful when a table name is not GraphQL compliant, like ``Users Address``.
```

An ``identifier`` like ``users_address`` will complement the ``"Users Address"``
table, so that it can be added to the GraphQL schema.

.. code-block:: http

   POST /v1/query HTTP/1.1
   Content-Type: application/json
   X-Hasura-Role: admin

   {
      "type": "track_table",
      "version": 2,
      "args": {
        "table": "Author Details",
        "configuration": {
           "identifier": "author_details"
        }
      }
   }

The GraphQL nodes and typenames
that are generated will be according to the ``identifier``. For example, in this case,
the nodes generated will be:

- ``users_address``
- ``users_address_one``
- ``users_address_aggregate``
- ``insert_users_address``
- ``insert_users_address_one``
- ``update_users_address``
- ``update_users_address_by_pk``
- ``delete_users_address``
- ``delete_users_address_by_pk``

.. note::

>   graphql-engine expects the constraint names(if any) of a table to be GraphQL compliant_ to be

Reviewer: ```suggestion
  graphql-engine requires the constraint names (if any) of a table to be GraphQL compliant_ in order to be
```

     - Required
     - Schema
     - Description

>    * - identifier

Reviewer 2: The name “identifier” here strikes me as very vague and a little confusing. We already use the phrase “custom fields” for the field names, why not call this a “custom name”?

Author: I have changed `identifier` to `custom name`

.. _set_table_custom_fields:

> set_table_custom_fields (deprecated)

Reviewer: Should we also say in [this table](https://deploy-preview-5719--hasura-docs.netlify.app/graphql/core/api-reference/schema-metadata-api/index.html#request-types) that this is depracated?

Author: I think yes, but would like to hear what @tirumaraiselvan also has to say

Reviewer 3: Doesn't it say already? https://github.com/hasura/graphql-engine/pull/5719/files#diff-e8eabc9c3c339a6063c7b2c29b6fdb009017d9e0459be2978aee7617f31c9afeR115

Or do you mean add an extra column called deprecated to the table? I don't think that is necessary at this point. Maybe when we have few more deprecated APIs.

Author: I added the `(deprecated)` indicator yesterday. Earlier, there was nothing. So, Marion wanted some sort of indicator that the API is deprecated

> -----------------------------------

Reviewer: ```suggestion
------------------------------------
```

> ``set_table_custom_fields`` has been deprecated now, use the

Reviewer: ```suggestion
``set_table_custom_fields`` has been deprecated. Use the
```

     - :ref:`CustomColumnNames`
     - Customise the column fields

.. _set_table_configuration:

set_table_configuration
-----------------------

> ``set_table_configuration`` sets the configuration of a table, through which

Reviewer: ```suggestion
``set_table_configuration`` sets the configuration of a table through which
```

> an identifier, custom root fields or custom column names of an already tracked

Reviewer: Is it only one of those (identifier, custom root fields, custom column names) that can be set? Or actually all three?

Author: Oh, we can set all the three. I'll make the change.

> table can be set. This will **replace** the already present configuration.

Reviewer: Can we link to the deprecated configuration here?

Author: Sure, I'll do that
